src/install/index.js

```diff
@@ -17,13 +25,42 @@ const keyFromDeployDoc = deployDoc => [
   deployDoc.build_info.version
 ].join(':');
 
+const appId = deployDoc => `_design/${deployDoc.build_info.application}`;
+
+const findDownloadedBuild = deployDoc => {
+  debug(`Locating already downloaded ${keyFromDeployDoc(deployDoc)}`);
+
+  const id = utils.getStagedDdocId(appId(deployDoc));
+
+  return DB.app.get(id, {
+      attachments: true,
+      binary: true
+  })
+    .catch(err => {
+      // Problem!
+      //
+      // We're running this because either we've already run this step, or we are "completing" an
+      // already staged build. So why doesn't the doc exist? Is this because we have an old or
+      // corrupt deployDoc? Is this because we've already moved this doc from staging to ready and a
+      // really late stage failed on us? We can't tell.
+      //
+      // However, we already had this problem. If you were attempting a complete action that failed
+      // before, and that failure was after you'd moved the staged ddoc into ready position that
+      // complete action would fail thereafter.
+
+      // TODO: think about what to do here
+      console.error('PROBLEMATIC ERROR!', err); //eslint-disable-line
```

Comment:
Your thoughts on this situation are as always, appreciated :-)

I am tempted to just keep this erroring out, and say you can't skip through deploys in situations where the staged ddoc has gone away.

Reply:
Absolutely error out. Maybe a better detailed message than 'PROBLEMATIC ERROR!' along with the 404 from Pouch.